webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultCsvValidationService.java

```diff
@@ -121,68 +134,91 @@ public DefaultCsvValidationService(final SchoolRepository schoolRepository) {
                 currentSchoolGroupYearSubject = null;
                 currentSchoolGroupYear = schoolGroupYear;
 
-                validateStudentCount(validationFailures, currentSchool, currentGroup, studentCount, recordNumber - 1);
-                studentCount = 0;
+                validateStudentCount(validationResults, currentSchool, currentGroup, studentCountPerGroup, recordNumber - 1);
+                studentCountPerGroup = 0;
                 currentGroup = group;
+                groupCountPerSchool++;
 
-                validateSchoolGroupYear(validationFailures, completedSchoolGroupYears, schoolGroupYear, recordNumber);
+                validateSchoolGroupYear(validationResults, completedSchoolGroupYears, schoolGroupYear, recordNumber);
             }
 
             //Validate the Subject
             if (currentSchoolGroupYearSubject == null) {
                 currentSchoolGroupYearSubject = subject;
             } else if (subject != null && !subject.equals(currentSchoolGroupYearSubject)) {
-                validationFailures.add(CsvValidationFailure.builder()
+                validationResults.add(CsvValidationResult.builder()
                         .row(recordNumber)
                         .message("Record is attempting to re-define group subject from [" + currentSchoolGroupYearSubject.name() + "] to [" + subject.name() + "]")
+                        .isOk(false)
                         .build());
+
                 currentSchoolGroupYearSubject = subject;
             }
 
             //Count students
             if (isNotBlank(record.get(HeaderSSID))) {
-                studentCount++;
+                studentCountPerGroup++;
             }
         }
-        validateStudentCount(validationFailures, currentSchool, currentGroup, studentCount, recordNumber);
+        validateStudentCount(validationResults, currentSchool, currentGroup, studentCountPerGroup, recordNumber);
+
+        if(allOk(validationResults)) {
+            validationResults.add(newSuccessValidationResult(recordNumber, groupCountPerSchool, getSchoolOrganization(availableSchools, currentSchool)));
+            return validationResults;
+        }
 
-        return validationFailures;
+        return  validationResults.stream().filter(x -> !x.isOk()).collect(Collectors.toList());
```

Comment:
So this method will only return all failures or all successes.  It will never return a mix of successes / failures because we don't have any kind of partial success.  
> If we go down we go down together.